app/access_policies/exercise_access_policy.rb

```diff
@@ -13,6 +13,9 @@ def self.action_allowed?(action, requestor, exercise)
     when :update, :destroy
       !exercise.is_published? && exercise.has_collaborator?(requestor) || \
       requestor.is_administrator?
+    when :new_version
+      exercise.is_published? && exercise.has_collaborator?(requestor) || \
+      requestor.is_administrator?
```

Comment:
This says an admin can make a new version even if the exercise is not published -- is that what you intended?